packages/js/test/plugins/auctionHouseModule/executePartialSale.test.ts

```diff
@@ -0,0 +1,261 @@
+import test, { Test } from 'tape';
+import { sol, token } from '@/types';
+import {
+  metaplex,
+  killStuckProcess,
+  createSft,
+  createWallet,
+  assertThrows,
+} from '../../helpers';
+import { createAuctionHouse } from './helpers';
+import { Keypair } from '@solana/web3.js';
+
+killStuckProcess();
+
+test('[auctionHouseModule] execute partial sale on an Auction House', async (t: Test) => {
+  // Given we have an Auction House and a SFT with 5 supply.
+  const mx = await metaplex();
+  const buyer = await createWallet(mx);
+  const auctionHouse = await createAuctionHouse(mx);
+  const sft = await createSft(mx);
+
+  await mx
+    .tokens()
+    .mint({
+      mintAddress: sft.address,
+      amount: token(5),
+    })
+    .run();
+
+  // And we listed that 5 SFTs for 1 SOL each.
+  const { listing } = await mx
+    .auctionHouse()
+    .list({
+      auctionHouse,
+      mintAccount: sft.address,
+      price: sol(5),
+      tokens: token(5),
+    })
+    .run();
+
+  // And we created a public bid on that SFT to buy only 3 Tokens.
+  const { bid } = await mx
+    .auctionHouse()
+    .bid({
+      auctionHouse,
+      buyer,
+      mintAccount: sft.address,
+      price: sol(3),
+      tokens: token(3),
+    })
+    .run();
+
+  // When we execute a sale with given listing and bid.
+  const { purchase } = await mx
+    .auctionHouse()
+    .executeSale({
+      auctionHouse,
+      listing,
+      bid,
+    })
+    .run();
+
+  // Then the user must receive 3 Tokens.
+  const buyerToken = await mx
+    .nfts()
+    .findByToken({ token: purchase.asset.token.address })
+    .run();
+
+  t.equal(buyerToken.token.amount.basisPoints.toNumber(), 3);
```

Comment:
Could we add more checks here?
- Checks on the state of the listing after the partial buy.
- Checks on the tokens owned by the seller.

Reply:
Not sure what to test on the listing's state, I believe nothing changes there. 

Added a check for the seller's balance.

Reply:
Hmm, how does the Program know that there are, say, two tokens left to sell on that listing? Because the seller could have hundreds of tokens for all we know but they only want to sell 5.

Do they use the `delegatedAmount` property of the token account? If so could we add a check on that?